src/app-sync.js

```diff
@@ -313,17 +314,17 @@ app.get('/download-user-file', async (req, res) => {
     return;
   }
 
-  let buffer;
-  try {
-    buffer = await fs.readFile(getPathForUserFile(fileId));
-  } catch (e) {
-    console.log(`Error: file does not exist: ${getPathForUserFile(fileId)}`);
-    res.status(500).send('File does not exist on server');
-    return;
-  }
-
   res.setHeader('Content-Disposition', `attachment;filename=${fileId}`);
-  res.send(buffer);
+  res.sendFile(getPathForUserFile(fileId), function (err) {
+    if (err) {
+      if (err instanceof HttpError && err.status == 404) {
+        console.log('File does not exist on server.');
+        res.status(500).send('File does not exist on server.');
+      } else {
+        next(err);
+      }
+    }
+  });
```

Comment:
I think it's totally fine to return 404 in this case. Makes the code easier :) 

Just don't remember to update the test file if you do this.

```suggestion
  res.sendFile(getPathForUserFile(fileId));
```

Reply:
Welp, that also resolves my problem with how to get type narrowing to work so that's nice :smile: